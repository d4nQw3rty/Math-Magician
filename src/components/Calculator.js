import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (event) => {
    const result = calculate(this.state, event.target.textContent);
    this.setState(result);
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div>
        <h1>Calculator</h1>
        <div className="gridContainer">
          <div className="parent">
            <div className="div1 height grayBackGround">
              <span className="total">
                { total }
                { operation }
                { next }
              </span>
            </div>
            <div className="div2 height"><button onClick={this.handleClick} type="button" name="AC" className="gridIlignItem  lightGrayBackGround">AC</button></div>
            <div className="div3 height"><button onClick={this.handleClick} type="button" name="+/-" className="gridIlignItem lightGrayBackGround">+ / -</button></div>
            <div className="div4 height"><button onClick={this.handleClick} type="button" name="%" className="gridIlignItem lightGrayBackGround">%</button></div>
            <div className="div5 height"><button onClick={this.handleClick} type="button" name="รท" className="gridIlignItem salmonBackGround">รท</button></div>
            <div className="div6 height">
              <button onClick={this.handleClick} type="button" name="7" className="gridIlignItem lightGrayBackGround">7</button>
              {' '}
            </div>
            <div className="div7 height">
              <button onClick={this.handleClick} type="button" name="8" className="gridIlignItem lightGrayBackGround">8</button>
              {' '}
            </div>
            <div className="div8 height">
              <button onClick={this.handleClick} type="button" name="9" className="gridIlignItem lightGrayBackGround">9</button>
              {' '}
            </div>
            <div className="div9 height">
              <button onClick={this.handleClick} type="button" name="x" className="gridIlignItem salmonBackGround">x</button>
              {' '}
            </div>
            <div className="div10 height">
              <button onClick={this.handleClick} type="button" name="4" className="gridIlignItem lightGrayBackGround">4</button>
              {' '}
            </div>
            <div className="div11 height">
              <button onClick={this.handleClick} type="button" name="5" className="gridIlignItem lightGrayBackGround">5</button>
              {' '}
            </div>
            <div className="div12 height">
              <button onClick={this.handleClick} type="button" name="6" className="gridIlignItem lightGrayBackGround">6</button>
              {' '}
            </div>
            <div className="div13 height">
              <button onClick={this.handleClick} type="button" name="-" className="gridIlignItem salmonBackGround">-</button>
              {' '}
            </div>
            <div className="div14 height">
              <button onClick={this.handleClick} type="button" name="1" className="gridIlignItem lightGrayBackGround">1</button>
              {' '}
            </div>
            <div className="div15 height">
              <button onClick={this.handleClick} type="button" name="2" className="gridIlignItem lightGrayBackGround">2</button>
              {' '}
            </div>
            <div className="div16 height">
              <button onClick={this.handleClick} type="button" name="3" className="gridIlignItem lightGrayBackGround">3</button>
              {' '}
            </div>
            <div className="div17 height">
              <button onClick={this.handleClick} type="button" name="+" className="gridIlignItem salmonBackGround">+</button>
              {' '}
            </div>
            <div className="div18 height">
              <button onClick={this.handleClick} type="button" name="0" className="gridIlignItem lightGrayBackGround">0</button>
              {' '}
            </div>
            <div className="div19 height">
              <button onClick={this.handleClick} type="button" name="." className="gridIlignItem lightGrayBackGround">.</button>
              {' '}
            </div>
            <div className="div20 height">
              <button onClick={this.handleClick} type="button" name="=" className="gridIlignItem salmonBackGround">=</button>
              {' '}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
