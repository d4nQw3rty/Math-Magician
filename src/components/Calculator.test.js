import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('Checking Calculator deployment', () => {
  test('if Calculator renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
