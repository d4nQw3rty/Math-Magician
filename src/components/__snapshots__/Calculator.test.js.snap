// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Calculator deployment if Calculator renders correctly 1`] = `
<div
  className="calculator"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="container"
  >
    <div
      className="gridContainer"
    >
      <div
        className="parent"
      >
        <div
          className="div1 height grayBackGround"
        >
          <span
            className="total"
          >
            0
          </span>
        </div>
        <div
          className="div2 height"
        >
          <button
            className="gridIlignItem  lightGrayBackGround"
            name="AC"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
        </div>
        <div
          className="div3 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
        </div>
        <div
          className="div4 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </div>
        <div
          className="div5 height"
        >
          <button
            className="gridIlignItem salmonBackGround"
            name="รท"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="div6 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
           
        </div>
        <div
          className="div7 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
           
        </div>
        <div
          className="div8 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
           
        </div>
        <div
          className="div9 height"
        >
          <button
            className="gridIlignItem salmonBackGround"
            name="x"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
           
        </div>
        <div
          className="div10 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
           
        </div>
        <div
          className="div11 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
           
        </div>
        <div
          className="div12 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
           
        </div>
        <div
          className="div13 height"
        >
          <button
            className="gridIlignItem salmonBackGround"
            name="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
           
        </div>
        <div
          className="div14 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
           
        </div>
        <div
          className="div15 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
           
        </div>
        <div
          className="div16 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
           
        </div>
        <div
          className="div17 height"
        >
          <button
            className="gridIlignItem salmonBackGround"
            name="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
           
        </div>
        <div
          className="div18 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
           
        </div>
        <div
          className="div19 height"
        >
          <button
            className="gridIlignItem lightGrayBackGround"
            name="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
           
        </div>
        <div
          className="div20 height"
        >
          <button
            className="gridIlignItem salmonBackGround"
            name="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
